### ISSUE TITLE
Windows 7

### ISSUE THREAD

**Issue by harryray33** [mpvnet-player/mpv.net#614]
Does this work with Win 7?

Thanks

**Comment by zhongfly** [comment]
mpv and libmpv.dll not support Windows 7 anymore, nothing mpv.net can do for this.

https://github.com/mpv-player/mpv?tab=readme-ov-file#system-requirements
> A not too ancient Linux, Windows 10 or later, or macOS 10.15 or later.

**Comment by Fran515** [comment]
![2023-12-15 20_44_43-mpv net](https://github.com/mpvnet-player/mpv.net/assets/59135036/11e6811a-3c35-479b-a381-543d880b75ef)
Same here,  mpv.net 7.0.0.3 doesn't run on Windows 7x64

**Comment by stax76** [comment]
I can update to .NET 8 then.

**Comment by therube** [comment]
> Does this work with Win 7?

Yes, but you need to make it work.
https://forum.doom9.org/showthread.php?p=1994994#post1994994

Older Beta, 7.0.0.1 & 7.0.0.2 do work, but more recent do not.
https://forum.doom9.org/showthread.php?p=1995227#post1995227

**Comment by Mikle-Bond** [comment]
So, to recap what was on doom9 forum:
- `mpv.net` works on windows 7 (as of current latest v7.1.1.0), but needs `libmpv-2.dll` to be replaced with the older one.
- The latest working `libmpv-2.dll` build is from 2023-09-17 (commit hash `181eddc`), the `x86-64` variant (as opposed to `x86-64-v3`, which also doesn't work). [Download link](https://sourceforge.net/projects/mpv-player-windows/files/libmpv/mpv-dev-x86_64-20230917-git-181eddc.7z/download).
- In the [diff](https://github.com/mpv-player/mpv/compare/181eddc..140d018#diff-2aa4b30edbbe4fdb9c34ba5df92b39cc3bd329a654aff809f0a1a716820370a3) between this build and the next one there are mentions of new `cplugin` system, and changes in Windows build process are recommended - it mentions setting `MPV_CPLUGIN_DYNAMIC_SYM` but it is nowhere to be seen in the build scripts. I don't see any other groundbreaking changes.

**Comment by vxiiduu** [comment]
> I can update to .NET 8 then.

Using VxKex allows mpv.net, as well as vanilla mpv, to work on Windows 7.
![image](https://github.com/mpvnet-player/mpv.net/assets/73044267/c06b5982-dd3e-4599-8840-adef962d8054)
Upgrading to .NET 8 would break this workaround.

**Comment by shirt-dev** [comment]
# Windows 7 support ended on January 14, 2020
Microsoft made a commitment to provide 10 years of product support for Windows 7 when it was released on October 22, 2009. This 10-year period has now ended, and Microsoft has discontinued Windows 7 support so that we can focus our investment on supporting newer technologies and great new experiences. The specific end of support day for Windows 7 was January 14, 2020. Technical assistance and software updates from Windows Update that help protect your PC are no longer available for the product. Microsoft strongly recommends that you move to Windows 11 to avoid a situation where you need service or support that is no longer available.

What does end of support mean for me?
> As of January 14, 2020, PCs running Windows 7 no longer receive security updates. Therefore, it's important that you upgrade to a modern operating system such as Windows 11, which can provide the latest security updates to help keep you and your data safer. In addition, Microsoft customer service is no longer available to provide Windows 7 technical support. Related services for Windows 7 are also being discontinued over time. For example, certain games such as Internet Backgammon and Internet Checkers as well as Electronic Program Guide for Windows Media Center are scheduled to be discontinued in January 2020.

What should I do?
> For most Windows 7 users, moving to a new device with Windows 11 is the recommended path forward. Today's PCs are faster, lightweight yet powerful, and more secure, with an average price that's considerably less than that of the average PC eight years ago. To find the best PC for you, [browse for compatible Windows 11 PCs](https://www.microsoft.com/windows/get-windows-11). 
Though we strongly recommend getting a new PC with Windows 11, we realize that is not always feasible, so we also suggest exploring an upgrade to Windows 10. 

Can I upgrade my existing PC to Windows 11?
> To take advantage of the latest hardware capabilities, we recommend moving to a [new PC](https://www.microsoft.com/windows/get-windows-11) with Windows 11, which brings a fresh look and new ways to stay creative and productive.

How can I upgrade to Windows 10 for free?
> The Windows 10 free upgrade offer ended on July 29, 2016. To upgrade to Windows 10, you will need to have a compatible PC and purchase [a full version of the software](https://www.microsoft.com/p/windows-10-home/d76qx4bznwk4?icid=CXSMC_w7eosfaq). We recommend that you don’t install Windows 10 on an older device, as some Windows 7 devices are not compatible with Windows 10 or could experience reduced feature availability. Before investing in a Windows 10 upgrade, please consider that Windows 10 will reach its end of support date on October 14, 2025. 

What happens if I continue to use Windows 7?
> If you continue to use Windows 7 now that support has ended, your PC will still work, but it will be more vulnerable to security risks and viruses. Your PC will continue to start and run, but will no longer receive software updates, including security updates, from Microsoft.

Can Windows 7 still be activated after January 14, 2020?
> Windows 7 can still be installed and activated after end of support; however, it will be more vulnerable to security risks and viruses due to the lack of security updates. After January 14, 2020, Microsoft strongly recommends that you use a newer version of Windows instead of Windows 7.

Will Internet Explorer still be supported on Windows 7?
> Support for Internet Explorer on a Windows 7 device was also discontinued on January 14, 2020. As a component of Windows, Internet Explorer follows the support lifecycle of the Windows operating system it’s installed on. See [Lifecycle FAQ – Internet Explorer for more information.](https://docs.microsoft.com/lifecycle/faq/internet-explorer-microsoft-edge)
. We recommend you use [Microsoft Edge](https://microsoftedgewelcome.microsoft.com/launch?url=https%3A%2F%2Faka.ms%2FMicrosoftSupport1). Microsoft Edge was built to bring you the best of the web, with more control and more privacy as you browse. 

Will Microsoft Security Essentials continue to protect my PC after end of support?
> Microsoft Security Essentials (MSE) will continue to receive signature updates after January 14, 2020. However, the MSE platform is no longer being updated. Click here to learn more about [learn more about Microsoft Security Essentials](https://support.microsoft.com/en-us/windows/what-is-microsoft-security-essentials-c25ad47a-7d15-8072-1438-b07dffcbbb20).

What if I'm running Windows 7 Enterprise?
> If you are using Windows as part of a work environment, we recommend you check first with your IT department or see [Windows 10 deployment support](https://docs.microsoft.com/windows/deployment/) to learn more.

What about Windows 7 Embedded?
> Windows for embedded devices such as ATMs or gas pumps have lifecycle dates that sometimes differ from versions of Windows that are used on PC devices.

Can I move my existing programs to a new Windows 11 PC?
> For the best possible experience, we recommend doing a fresh installation of your programs and apps on your new Windows 11 PC. Microsoft Store has a variety of third-party apps that can be easily downloaded and installed. On your Windows 11 PC, you can open Microsoft Store by selecting the icon on the taskbar or by searching for "Microsoft Store" in the search box on the taskbar.
 For any software not currently available in Microsoft Store, we recommend going to that company's official website and downloading the Windows 11 version from there. In limited circumstances, some older software may not have an updated version that's compatible with Windows 11.

How will Windows 7 end of support affect my Microsoft Office apps?
> Your experience may be different depending on the Office version you're running. To learn more, see [Windows 7 end of support and Office](https://support.office.com/article/78f20fab-b57b-44d7-8368-06a8493f3cb9?wt.mc_id=OfficeEOL_%20XlinkSMC4057281). It's important to note that Windows 7 itself no longer receives security updates, leaving your device vulnerable to security threats. As such, if you're running Microsoft Office on a Windows 7 PC, we recommend you [move to a Windows 11 PC](https://support.microsoft.com/en-us/windows/getting-ready-for-the-windows-11-upgrade-eb50813f-c7da-4cf8-89a3-6ba0d33b2773).

Why did I receive a notification about Windows 7?
> Support for Windows 7 came to an end on January 14, 2020. You are receiving notifications as a Windows 7 customer to remind you that your device is no longer supported and no longer receiving security updates.  We recommend moving to a [new PC](https://support.microsoft.com/en-us/windows/getting-ready-for-the-windows-11-upgrade-eb50813f-c7da-4cf8-89a3-6ba0d33b2773) with Windows 11.  More information on the notification is available [here](https://support.microsoft.com/en-us/topic/you-received-a-notification-your-windows-7-pc-is-out-of-support-3278599f-9613-5cc1-e0ee-4f81f623adcf).

**Comment by harryray33** [comment]
Thanks, mickle-bond, vxiiduu and the rube.
I'll give the various suggestions a try out to see if it works with Win 7.

**Comment by Mikle-Bond** [comment]
@harryray33 yes, I tried VxKex, seems to work for me too. The setting didn't persist when I updated the mpvnet.portable with Chocolatey though.

@shirt-dev thanks for the great paste, really good stuff, very like scrolling past it.

**Comment by harryray33** [comment]
@mikle-bond @vxiiduu

I had a look at VxKex, seems like a good idea, but I got over 30 hits on it from Virustotal.

**Comment by harryray33** [comment]
Is VxKex safe to use?

**Comment by Mikle-Bond** [comment]
@harryray33 well, my PC didn't blow up in the past month of using it. So, I'd say it is exactly *as safe* as using a discontinued OS itself.
